src/Service/NuRaftStateMachine.cpp

```diff
@@ -753,9 +753,13 @@ bool NuRaftStateMachine::existSnapshotObject(snapshot & s, ulong obj_id)
 
 bool NuRaftStateMachine::apply_snapshot(snapshot & s)
 {
-    /// TODO: double buffer load or multi thread load
+    /// If the invoker is from NuRaft, we should reset the state machine
+    LOG_INFO(log, "reset state machine.");
+    reset();
```

Comment:
Should we get `snapshot_mutex` before reset?

Reply:
Fixed thanks